### PR TITLE
Reissue used before declaration error for const enums in `isolatedModules`

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/57173

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -3924,6 +3924,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             else if (result.flags & SymbolFlags.RegularEnum) {
                 diagnosticMessage = error(errorLocation, Diagnostics.Enum_0_used_before_its_declaration, declarationName);
             }
+            else {
+                Debug.assert(!!(result.flags & SymbolFlags.ConstEnum));
+                if (compilerOptions.isolatedModules) {
```
Comment (andrewbranch) [inline src/compiler/checker.ts:3929]:
```suggestion
                if (getIsolatedModules(compilerOptions)) {
```

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: src/compiler/checker.ts (hunk 1 above)
Comment (andrewbranch) [reply src/compiler/checker.ts:3929]: I would commit this suggestion for you and approve, but GitHub can't handle this on checker.ts 🥲

Comment (Andarist) [reply src/compiler/checker.ts:3929]: 🫠

I pushed out the change + the relevant missing test case for `verbatimModuleSyntax` :p

**Review by andrewbranch — APPROVED** [review]
Thanks!

**Pull request merged**